### PR TITLE
Tests for array.js

### PR DESCRIPTION
**Pull request by hrishikeshs**
I've added a couple of tests for array.js, I'll need to add a lot more, especially for invalid inputs and stuff. This PR is just to get feedback and facilitate discussion around what sort of failure strategies we want and what kind of tests should be written and possibly, what features need to be added/removed to array.js.

### CODE REVIEW

File: tests/array_test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  });
+  
+  it('returns the array with last element popped', function() {
+    var arr = [1,2,3,4];
+    var arr_len = arr.length;
+    var popped_array = array.drop(arr);
+    popped_array.length.should.equal(arr_len -1);
+    array.is(popped_array).should.equal(true);
+    popped_array.should.equal(arr); //test they are both the same reference
+    popped_array[0].should.equal(1);
```
Comment (hrishikeshs) [inline tests/array_test.js:34]: would be nice if we had `array.equivalent(a,b)` to return true if the elements are all the same. But of course, the question of whether it is a shallow comparison / deep comparison is a big one. All because we have an assignment statement in JavaScript. :smiling_imp:

**Comment by hrishikeshs** [comment]
Needs test cases for invalid inputs, empty arguments and so on.

**Comment by hrishikeshs** [comment]
@rtm If you are okay with the way the tests are written, I think this is good to merge. I am planning to write test cases for happy paths on all files and then handle all erroneous inputs later so I get a overview of the whole codebase.
